### PR TITLE
Relational

### PR DESCRIPTION
**Pull request by William-TogetherScience**

### CODE REVIEW

File: ts-port/core/basic.ts [hunk 1]
```diff
@@ -123,6 +121,20 @@ const _Basic = (superclass: any) => class _Basic extends superclass {
...
     }
 
+    isinstance(cls: any) {
+        if (this instanceof cls) {
+            return true;
+        }
+        const supers = (this.constructor as any).supers;
+        if (supers) {
+            if (cls.clsname) {
+                return supers.has(cls.clsname);
```
Comment (William-TogetherScience) [inline ts-port/core/basic.ts]: figure out adding class name to supers

File: ts-port/core/basic.ts [hunk 2]
```diff
@@ -202,22 +209,24 @@ const _Basic = (superclass: any) => class _Basic extends superclass {
         }
         const n1 = self.constructor.name;
         const n2 = other.constructor.name;
-        if (n1 && n2) {
-            return (n1 > n2 as unknown as number) - (n1 < n2 as unknown as number);
+        let c = (n1 > n2 as unknown as number) - (n1 < n2 as unknown as number)
```
Comment (William-TogetherScience) [inline ts-port/core/basic.ts:213]: comments

File: ts-port/core/expr.ts [hunk 3]
```diff
@@ -172,6 +172,22 @@ const Expr = (superclass: any) => class Expr extends mix(superclass).with(_Basic
     as_numer_denom() {
         return [this, S.One];
     }
+
+    __ge__(other: any) {
```
Comment (William-TogetherScience) [inline ts-port/core/expr.ts:176]: is any the correct type??

File: ts-port/core/relational.ts [hunk 4]
```diff
@@ -155,38 +797,184 @@ export function is_ge(lhs: any, rhs: any) {
...
+    >>> is_eq(a, b)
+    True
+    >>> is_neq(a, b)
+    False
+    */
+
+    const retval = _eval_is_eq(lhs, rhs)
     if (typeof retval !== "undefined") {
         return retval;
-    } else {
```
Comment (William-TogetherScience) [inline ts-port/core/relational.ts:166]: what happened?

File: ts-port/core/relational.ts [hunk 5]
```diff
@@ -155,38 +797,184 @@ export function is_ge(lhs: any, rhs: any) {
...
+        if (fuzzy_and([lhs.is_extended_real(), rhs.is_extended_real()])) {
+            return fuzzy_xor([lhs.is_extended_positive(), fuzzy_not(rhs.is_extended_positive())]);
+        }
+
 
-        // SKIPPING ASSUMPTIONS WRAPPER FOR NOW
+        // complex numbers not yet supported
+    }
 
-        if (lhs.is_extended_real() && rhs.is_extended_real()) {
```
Comment (William-TogetherScience) [inline ts-port/core/relational.ts:179]: ??

File: ts-port/core/relational.ts [hunk 6]
```diff
@@ -155,38 +797,184 @@ export function is_ge(lhs: any, rhs: any) {
...
+        const dif = lhs.__sub__(rhs);
+        const z = dif.is_zero();
+        if (typeof z !== "undefined") {
+            if (!z && dif.is_commutative()) {
+                return false;
+            }
+            if (z) {
                 return true;
             }
-            const diff = new Add(true, true, lhs, new Mul(true, true, rhs, S.NegativeOne));
```
Comment (William-TogetherScience) [inline ts-port/core/relational.ts:183]: ??

**Review by gunnarmein-ts — APPROVED** [review]

**Pull request merged**